sql/2020/20_Caching/resource_without_cache.sql

```diff
@@ -0,0 +1,30 @@
+#standardSQL
+# Resources served without using cache
+SELECT
+  client,
+  COUNT(0) AS total_requests,
+  type,
+  COUNTIF(not_cacheable) AS total_not_cacheable,
+  COUNTIF(NOT not_cacheable AND uses_cache) AS total_using_cache,
+  COUNTIF(NOT not_cacheable AND NOT uses_cache) AS total_not_using_cache,
+
+  ROUND(COUNTIF(not_cacheable) * 100 / COUNT(0), 2) AS perc_not_cacheable,
+  ROUND(COUNTIF(NOT not_cacheable AND uses_cache) * 100 / COUNT(0), 2) AS perc_using_cache,
+  ROUND(COUNTIF(NOT not_cacheable AND NOT uses_cache) * 100 / COUNT(0), 2) AS perc_not_using_cache
+FROM (
+  SELECT
+    client,
+    type,
+    REGEXP_CONTAINS(resp_cache_control, r'(?i)(no-cache|no-store)') AS not_cacheable,
+    expAge > 0 AS uses_cache
+  FROM
+    `httparchive.almanac.requests`
+  WHERE
+    date = '2020-08-01'
+)
+GROUP BY
+  client,
+  type
+ORDER BY
+  type,
+  client

```

Comment:
Nit, but personally I prefer to have `GROUP BY` and `ORDER BY` fields in same order (and ideally in same order as they appear in `SELECT` clause) to spot errors and/or add and remove columns.

(same applies to other queries)

Reply:
Updated the query. Following same convention in other queries also.